Site/admin/components/Account/MergeConfirm.php

```diff
@@ -0,0 +1,258 @@
+<?php
+
+require_once 'Admin/pages/AdminDBConfirmation.php';
+require_once 'SwatDB/SwatDB.php';
+require_once 'SwatDB/SwatDBClassMap.php';
+
+/**
+ * Account merge confirmation page
+ *
+ * @package   Site
+ * @copyright 2016 silverorange
+ */
+abstract class SiteAccountMergeConfirm extends AdminDBConfirmation
+{
+	// {{{ protected properties
+
+	/**
+	 * @var integer
+	 */
+	protected $id;
+
+	/**
+	 * @var SiteAccount
+	 */
+	protected $account;
+
+	/**
+	 * @var integer
+	 */
+	protected $id2;
+
+	/**
+	 * @var SiteAccount
+	 */
+	protected $account2;
+
+	/**
+	 * @var boolean
+	 */
+	protected $keep_first;
+
+	// }}}
+
+	// init phase
+	// {{{ protected function initInternal()
+
+	protected function initInternal()
+	{
+		parent::initInternal();
+
+		$this->id = SiteApplication::initVar('id');
+		$this->account = $this->getAccount($this->id, $this->account);
+
+		$this->id2 = SiteApplication::initVar('id2');
+		$this->account2 = $this->getAccount($this->id2, $this->account2);
+
+		$this->keep_first = SiteApplication::initVar('keep_first');
+	}
+
+	// }}}
+	// {{{ protected function getAccount()
+
+	protected function getAccount($id, $account)
```

Comment:
Passing the $account here is not necessary since this is only called once during initInternal() and getSourceAccount/getTargetAccount are used elsewhere.